dffml/source/json.py

```diff
@@ -20,13 +20,22 @@ class JSONSource(FileSource, MemorySource):
     async def load_fd(self, fd):
         repos = json.load(fd)
         self.mem = {
-            src_url: Repo(src_url, data=data)
-            for src_url, data in repos.items()
+            self.config.label: [
+                {
+                    src_url: Repo(src_url, data=data)
+                    for src_url, data in repos.items()
+                }
+            ]
         }
```

Comment:
You're syntax here is incorrect `{ variable: [] }` is invalid. Only strings can go inside dictionary definitions. What you were doing is valid in JavaScript, but not python :)

You'll need to define this at the top

```python
class MissingLabel(Exception):
    pass
```

Then, `load_fd` should be along the lines of:

```python
        repos = json.load(fd)
        if not self.config.label in repos:
            raise MissingLabel('%r not found in %s', self.config.label, ', '.join(repos.keys()))
        self.mem = {
            src_url: Repo(src_url, data=data)
            for src_url, data in repos[self.config.label].items()
        }
```